lib/daru/io/importers/redis.rb

```diff
@@ -0,0 +1,195 @@
+        #   df
+        #
+        #   #=> <Daru::DataFrame(4x2)>
+        #   #         name     age
+        #   #    0  Tyrion      32
+        #   #    1   Jamie      37
+        #   #    2  Cersei      37
+        #   #    3 Joffrey      19
+        #
+        # @example Querying for matching keys with count and offset
+        #   # Say, the Redis connection has this setup
+        #   # Key "key:1" => { "name" => "name1", "age" => "age1" }.to_json
+        #   # Key "key:2" => { "name" => "name2", "age" => "age2" }.to_json
+        #   # Key "key:3" => { "name" => "name3", "age" => "age3" }.to_json
+        #   # ...
+        #   # Key "key:2000" => { "name" => "name2000", "age" => "age2000" }.to_json
+        #
+        #   connection = {url: "redis://:[password]@[hostname]:[port]/[db]"}
+        #   Daru::DataFrame.from_redis(connection, match: "key:1*")
+        #
+        #   #=> #<Daru::DataFrame(1111x2)>
+        #   #              name      age
+        #   # key:1045 name1045  age1045
+        #   # key:1919 name1919  age1919
+        #   # key:1155 name1155  age1155
+        #   # key:1649 name1649  age1649
+        #   #      ...      ...      ...
+        #
+        #   Daru::DataFrame.from_redis({}, match: "key:1*", offset: 400, count: 200)
+        #
+        #   #=> #<Daru::DataFrame(200x2)>
+        #   #              name      age
+        #   # key:1927 name1927  age1927
+        #   # key:1759 name1759  age1759
+        #   # key:1703 name1703  age1703
+        #   # key:1640 name1640  age1640
+        #   #   ...        ...      ...
+        #
+        #   Daru::DataFrame.from_redis({}, match: "key:1*", offset: 1100, count: 200)
+        #   #=> #<Daru::DataFrame(12x2)>
+        #   #              name      age
+        #   # key:1084 name1084  age1084
+        #   # key:1195 name1195  age1195
+        #   # key:1250 name1250  age1250
+        #   # key:1303 name1303  age1303
+        #   # key:1341 name1341  age1341
+        #   #   ...        ...      ...
+        def initialize(connection={}, *keys, match: nil, count: nil,
+          offset: 1)
+          @match  = match
+          @count  = count
+          @offset = offset-1
+          @client = get_client(connection)
+          @keys   = choose_keys(*keys).map(&:to_sym)
+        end
+
+        def call
+          vals = @keys.map { |key| ::JSON.parse(@client.get(key), symbolize_names: true) }
+          Util.guess_parse(@keys, vals)
+        end
+
+        private
+
+        def choose_keys(*keys)
+          if keys.count.zero?
+            cursor = nil
+
+            # Loop to iterate through paginated results of Redis#scan.
+            until cursor == '0'
+              response = @client.scan(cursor, match: @match, count: @count)
+              cursor, chunk = response
+              keys.concat(chunk).uniq!
+              return keys[@offset..@offset+@count-1] unless @count.nil? || keys.count < (@offset+@count)
+            end
+            keys[@offset..-1]
```

Comment:
This can be DRY-ed up:
```ruby
until cursor == '0' || (max_offset && keys.count > max_offset)
  cursor, chunk = @client.scan(cursor, match: @match, count: @count)
  keys.concat(chunk).uniq!
end
keys[offset..(max_offset || -1)]
```